src/Psl/DateTime/DateTime.php

```diff
@@ -0,0 +1,552 @@
+        $calendar = IntlCalendar::createInstance(
+            Internal\to_intl_timezone($timezone),
+        );
+
+        $calendar->setTime(
+            $timestamp->getSeconds() * MILLISECONDS_PER_SECOND,
+        );
+
+        $year = $calendar->get(IntlCalendar::FIELD_YEAR);
+        $month = $calendar->get(IntlCalendar::FIELD_MONTH) + 1;
+        $day = $calendar->get(IntlCalendar::FIELD_DAY_OF_MONTH);
+        $hour = $calendar->get(IntlCalendar::FIELD_HOUR_OF_DAY);
+        $minute = $calendar->get(IntlCalendar::FIELD_MINUTE);
+        $second = $calendar->get(IntlCalendar::FIELD_SECOND);
+        $nanoseconds = $timestamp->getNanoseconds();
+
+        /** @psalm-suppress MissingThrowsDocblock */
+        return new static(
+            $timezone,
+            $timestamp,
+            $year,
+            $month,
+            $day,
+            $hour,
+            $minute,
+            $second,
+            $nanoseconds,
+        );
+    }
+
+
+    /**
+     * Parses a date and time string into an instance of {@see Timestamp} using a specific format pattern, with optional customization for timezone and locale.
+     *
+     * This method is specifically designed for cases where a custom format pattern is used to parse the input string.
+     *
+     * It allows for precise control over the parsing process by specifying the exact format pattern that matches the input string.
+     *
+     * Additionally, the method supports specifying a timezone and locale for parsing, enabling accurate interpretation of locale-specific formats.
+     *
+     * Example usage:
+     *
+     * ```php
+     * $raw_string = '2023-03-15 12:00:00';
+     * $parsed_datetime = DateTime\DateTime::parse($raw_string, 'yyyy-MM-dd HH:mm:ss', DateTime\Timezone::Utc, Locale\Locale::English);
+     * ```
+     *
+     * @param string $raw_string The date and time string to parse.
+     * @param null|FormatPattern|string $pattern The custom format pattern for parsing the date and time. If null, uses a default pattern.
+     * @param null|Timezone $timezone Optional timezone for parsing. If null, uses the system's default timezone.
+     * @param null|Locale $locale Optional locale for parsing. If null, uses the system's default locale.
+     *
+     * @throws Exception\RuntimeException If the parsing process fails.
+     *
+     * @return static Returns an instance of {@see Timestamp} representing the parsed date and time.
+     *
+     * @see https://unicode-org.github.io/icu/userguide/format_parse/datetime/#datetime-format-syntax
+     * @see TemporalInterface::format()
+     *
+     * @psalm-mutation-free
+     */
+    public static function parse(string $raw_string, null|FormatPattern|string $pattern = null, ?Timezone $timezone = null, null|Locale $locale = null): static
+    {
+        $timezone ??= Timezone::default();
+
+        return self::fromTimestamp(Timestamp::parse($raw_string, $pattern, $timezone, $locale), $timezone);
+    }
+
```

Comment:
Currently you have both parse which takes a pattern and fromString which takes some kind of less enforced date/time-style. However it always has to be in some kind of format. Would it make sense to add one more - in line of [strtotime](https://www.php.net/manual/en/function.strtotime.php). The alternative is to use `DateTime::fromTimestamp(Timestamp::fromParts(strtotime(...))` which maybe becomes a bit much? A similar method could be added to `Timestamp`

Reply:
the reason i didn't add it is that it isn't a think really outside of PHP as far as i seen, most other languages do not have such features, and thinking about it, i couldn't find a case where you would have a datetime string that you are unaware of its format.